lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -55,14 +55,19 @@ namespace Js
     template<bool zero>
     bool JavascriptConversion::SameValueCommon(Var aLeft, Var aRight)
     {
+        if (aLeft == aRight)
+        {
+            return true;
+        }
+
         TypeId leftType = JavascriptOperators::GetTypeId(aLeft);
-        TypeId rightType = JavascriptOperators::GetTypeId(aRight);
 
         if (JavascriptOperators::IsUndefinedOrNullType(leftType))
         {
-            return leftType == rightType;
+            return false;
```

Comment:
What happens if both values are undefined but belonged to different script context?

Reply:
I would have thought they should be marshalled to correct script context by here? If that's true, then Booleans are broken already in the existing code.